### ISSUE TITLE
[Bug]: The thumbnail is cropped in some cases in a monitored folder.

### ISSUE THREAD

**Issue by joaochris** [sircharlo/meeting-media-manager#3459]
### What happened?

In some cases the thumbnail is cut off, it does not fill the entire square. I tested with videos from the circuit overseer's media that are in the restricted area. There is inconsistency, sometimes it fills the entire square and other times it doesn't. :D

![388283237-53dfd940-81ac-473d-8ad2-e7bf3b2e8adc](https://github.com/user-attachments/assets/12ad04d2-d00f-48e4-8021-ad2a826963a2)

### To Reproduce

....

### What did you expect to happen?

.....

### Version

Latest version

### What platform are you seeing the problem on?

Windows

### Confirmations

- [X] I confirm that I have read the documentation website.
- [X] I confirm that I have searched through the existing issues to make sure my bug report is not a duplicate.
- [X] I confirm that I have searched through the existing discussions to make sure my bug report is not a duplicate.
- [X] If I am using macOS, I confirm that I have executed the Terminal commands found in the <a href="https://sircharlo.github.io/meeting-media-manager/fr/using-at-a-kingdom-hall#additional-steps-for-macos-users" target="_blank">additional steps for macOS users</a> found on the documentation website before submitting this bug report.

**Comment by sircharlo** [comment]
What exact steps do you follow that lead to the "incorrect" thumbnail?
When you see the "incorrect" thumbnail, please:
- open the Developer Tools (CTRL-SHIFT-I, 3 times)
- Go to `Application` > `Local storage`
- Expand `jw-store`
- Right-click on `additionalMediaMaps` > `Copy object`
- Paste the value here in a code block, and indicate which media item is at fault

Thanks for helping!

**Comment by joaochris** [comment]
@sircharlo 

`{
    "87643d0c-efea-41e9-a095-69bae3a9ef5b": {
        "2024/11/24": []
    }
}`
![image](https://github.com/user-attachments/assets/9d247a10-2da4-4577-af94-03c10503ecc7)

Like this?

**Comment by sircharlo** [comment]
Is this example video:

- Added from playlist file?
- Added from a JWPUB file?
- Already in the watched folder?

**Comment by joaochris** [comment]
> Is this example video:
> 
> - Added from playlist file?
> - Added from a JWPUB file?
> - Already in the watched folder?

I only added the video to the monitored folder.

**Comment by sircharlo** [comment]
> > Is this example video:
> > 
> > * Added from playlist file?
> > * Added from a JWPUB file?
> > * Already in the watched folder?
> 
> I only added the video to the monitored folder.

Videos from a watched folder will always have the same thumbnail "look". 

Could it be that you're comparing the thumbnail of a video you added from the Video menu or from a JWPUB file, to one that was added from the watcher? 

Those will always look different.

**Comment by joaochris** [comment]
> > > Is this example video:
> > > 
> > > * Added from playlist file?
> > > * Added from a JWPUB file?
> > > * Already in the watched folder?
> > 
> > I only added the video to the monitored folder.
> 
> Videos from a watched folder will always have the same thumbnail "look". 
> 
> Could it be that you're comparing the thumbnail of a video you added from the Video menu or from a JWPUB file, to one that was added from the watcher? 
> 
> Those will always look different.

No, from the monitored folder, this happens with some videos, in some the thumbnail is completely filled, and in others it is cut off. I'll send you the video later, I think you'll understand better

**Comment by sircharlo** [comment]
> > > > Is this example video:
> > > > 
> > > > * Added from playlist file?
> > > > * Added from a JWPUB file?
> > > > * Already in the watched folder?
> > > 
> > > 
> > > I only added the video to the monitored folder.
> > 
> > 
> > Videos from a watched folder will always have the same thumbnail "look".
> > Could it be that you're comparing the thumbnail of a video you added from the Video menu or from a JWPUB file, to one that was added from the watcher?
> > Those will always look different.
> 
> No, from the monitored folder, this happens with some videos, in some the thumbnail is completely filled, and in others it is cut off. I'll send you the video later, I think you'll understand better

Sure, send one video that looks "normal", and one that doesn't. 
Thanks!

**Comment by joaochris** [comment]
> > > > > Is this example video:
> > > > > 
> > > > > * Added from playlist file?
> > > > > * Added from a JWPUB file?
> > > > > * Already in the watched folder?
> > > > 
> > > > 
> > > > I only added the video to the monitored folder.
> > > 
> > > 
> > > Videos from a watched folder will always have the same thumbnail "look".
> > > Could it be that you're comparing the thumbnail of a video you added from the Video menu or from a JWPUB file, to one that was added from the watcher?
> > > Those will always look different.
> > 
> > No, from the monitored folder, this happens with some videos, in some the thumbnail is completely filled, and in others it is cut off. I'll send you the video later, I think you'll understand better
> 
> Sure, send one video that looks "normal", and one that doesn't. 
> Thanks!
> 

@sircharlo 

https://drive.google.com/drive/folders/14MgAaEXcjVQjPiYDrR-iiXhZ3I1dhjhC

In this playlist file there is a video with the normal thumbnail and another with the cropped one :)

**Comment by sircharlo** [comment]
The issue is with the types of videos.
One video is _embedded_ in the playlist. This file could be any video, from any source. The thumbnail is extracted from the embedded thumbnail in the video file's metadata, if present. If not, a thumbnail is generated automatically from the video itself. In most cases, the embedded thumbnail is square.

The other video is _not embedded_ in the playlist; it's a 'link' to the actual video on the site. (In this case, it's a link to [jwbvod24_45](https://www.jw.org/en/library/videos/#en/mediaitems/pub-jwbvod24_45_VIDEO). You can make a playlist with dozens of videos that are not embedded; the resulting playlist will have a tiny filesize, because it does not contain the video files themselves, but rather links to them. The app (whether JWL or MMM) will download the files as needed once the playlist is opened/imported. 
Thumbnails for videos that are _not embedded_ can be retrieved at the same time as the videos themselves are downloaded. These thumbnails are in a 16:9 format, and generally look nicer. If this lookup fails, or if no thumbnail exists, then a thumbnail is extracted from the embedded thumbnail in the video file's metadata, if present. If not, a thumbnail is generated automatically from the video itself.

I hope this explains what is happening and why it must be this way! 😄

**Comment by joaochris** [comment]
I understood. Then: 

1. For embedded videos, thumbnails are usually square in shape, extracted from metadata, or automatically generated. 

2. For linked videos, thumbnails usually have a 16:9 aspect ratio, but may vary if the information search fails. 

**What if you automatically resize all thumbnails to a fixed aspect ratio (like 16:9) regardless of their origin?**

**Comment by sircharlo** [comment]
> I understood. Then:
> 
> 1. For embedded videos, thumbnails are usually square in shape, extracted from metadata, or automatically generated.
> 2. For linked videos, thumbnails usually have a 16:9 aspect ratio, but may vary if the information search fails.

Correct!

> **What if you automatically resize all thumbnails to a fixed aspect ratio (like 16:9) regardless of their origin?**

What would be the advantage of this?

**Comment by joaochris** [comment]
> What would be the advantage of this?


The advantage of resizing all thumbnails to a fixed aspect ratio, like 16:9, would be to improve the visual appearance and consistency of the layout. This doesn't change the functionality, but it makes the experience more visually pleasant :). However, if you think this doesn't make sense, that's okay 😃.

**Comment by mtdvlpr** [comment]
> The advantage of resizing all thumbnails to a fixed aspect ratio, like 16:9, would be to improve the visual appearance and consistency of the layout. This doesn't change the functionality, but it makes the experience more visually pleasant :). However, if you think this doesn't make sense, that's okay 😃.

If you force resize a square thumbnail to 16:9, it will get distorted. It won't look nice anymore.

**Comment by sircharlo** [comment]
Yeah, let's shelve this one for now. I don't see a functional reason to implement a change here.
